Spigot-Server-Patches/0491-Implement-JellySquid-s-Entity-Collision-optimisation.patch

```diff
@@ -0,0 +1,397 @@
+     // Paper end - optimise hard collision
+ 
+     default Stream<VoxelShape> b(@Nullable Entity entity, AxisAlignedBB axisalignedbb, Set<Entity> set) {
++        //Paper start - use fast entity collisions
++        return me.jellysquid.mods.lithium.common.shapes.LithiumEntityCollisions.getEntityCollisions(this, entity, axisalignedbb, set);
++        /*
++        //Paper end
+         if (axisalignedbb.a() < 1.0E-7D) {
+             return Stream.empty();
+         } else {
+@@ -66,6 +70,8 @@ public interface IEntityAccess {
+ 
+             return stream.filter(axisalignedbb1::c).map(VoxelShapes::a);
+         }
++         */ //Paper - comment out unused code
+     }
+ 
+     @Nullable
+diff --git a/src/main/java/net/minecraft/server/VoxelShapes.java b/src/main/java/net/minecraft/server/VoxelShapes.java
+index 08c83c62d..802900e34 100644
+--- a/src/main/java/net/minecraft/server/VoxelShapes.java
++++ b/src/main/java/net/minecraft/server/VoxelShapes.java
+@@ -20,11 +20,11 @@ public final class VoxelShapes {
+     });
+     public static final VoxelShape a = create(Double.NEGATIVE_INFINITY, Double.NEGATIVE_INFINITY, Double.NEGATIVE_INFINITY, Double.POSITIVE_INFINITY, Double.POSITIVE_INFINITY, Double.POSITIVE_INFINITY);
+     private static final VoxelShape c = new VoxelShapeArray(new VoxelShapeBitSet(0, 0, 0), new DoubleArrayList(new double[]{0.0D}), new DoubleArrayList(new double[]{0.0D}), new DoubleArrayList(new double[]{0.0D}));
+-
++    public static final VoxelShape empty() {return a();} //Paper - OBFHELPER
+     public static VoxelShape a() {
+         return VoxelShapes.c;
+     }
+-
++    public static final VoxelShape fullcube() {return b();} //Paper - OBFHELPER
+     public static VoxelShape b() {
+         return VoxelShapes.b;
+     }
+@@ -129,6 +129,7 @@ public final class VoxelShapes {
+     }
+ 
+     public static final boolean applyOperation(VoxelShape voxelshape, VoxelShape voxelshape1, OperatorBoolean operatorboolean) { return VoxelShapes.c(voxelshape, voxelshape1, operatorboolean); } // Paper - OBFHELPER
++    public static final boolean matchesAnywhere(VoxelShape voxelshape, VoxelShape voxelshape1, OperatorBoolean operatorboolean) { return c(voxelshape, voxelshape1, operatorboolean);} // Paper - OBFHELPER
+     public static boolean c(VoxelShape voxelshape, VoxelShape voxelshape1, OperatorBoolean operatorboolean) {
+         if (operatorboolean.apply(false, false)) {
+             throw (IllegalArgumentException) SystemUtils.c(new IllegalArgumentException());
+diff --git a/src/main/java/net/minecraft/server/WorldBorder.java b/src/main/java/net/minecraft/server/WorldBorder.java
+index 5f10c4338..81636c2ba 100644
+--- a/src/main/java/net/minecraft/server/WorldBorder.java
++++ b/src/main/java/net/minecraft/server/WorldBorder.java
+@@ -48,6 +48,8 @@ public class WorldBorder {
+         return this.b(entity.locX(), entity.locZ());
+     }
+ 
++    public final VoxelShape asVoxelShape(){ return a();} //Paper - OBFHELPER
```

Comment:
need to move this down 1 line so it borders.
that single line can be enough to throw off patch apply conflicts (which we want!)